app/jobs/descriptive_metadata_import_job.rb

```diff
@@ -23,6 +23,7 @@ def perform(bulk_action_id, params)
                        .bind { |description| validate_changed(cocina_object, description) }
                        .bind { |description| open_version(cocina_object, description) }
                        .bind { |description, new_cocina_object| CocinaValidator.validate_and_save(new_cocina_object, description:) }
+                       .bind { VersionService.close(identifier: cocina_object.externalIdentifier) }
```

Comment:
Since `VersionService.close` doesn't return a promise, I don't think you can chain this result here, can you?  Can you write a test for `VersionService.close` raising `UnexpectedResponse`?  Does it call the failure lambda?

Reply:
can you put those two items in the same block?

Reply:
Yes, tried that locally and it does work, only downside I can see is that it makes the calls less atomic (i.e. update can succeeed, close can fail).  Not sure that has any real world implications though.

Reply:
Just pushed a commit with this change.  Though would still like to understand why it wasn't working before.

Reply:
I'm guessing that this line in the code is returning a Success or Failure without any return: https://github.com/sul-dlss/argo/blob/main/app/services/cocina_validator.rb#L17

while this method (for example), is returning a Success with data structures returned which are then available to pass down the chain: https://github.com/sul-dlss/argo/blob/main/app/jobs/descriptive_metadata_import_job.rb#L44

So I guess I'm trying to understanding how `to_result` works